### ISSUE TITLE
Studio can't interact with iOS native pop-up when landscape

### ISSUE THREAD

**Issue by klyovan** [mobile-dev-inc/maestro#2065]
### Is there an existing issue for this?

- [X] I have searched the existing issues and didn't find mine.

### Steps to reproduce

1. Run `maestro studio`  on flutter app builded for ios
2. Invoke native pop-up using Platfrom Channels (or `signIn` method from [google sign in plugin](https://pub.dev/packages/google_sign_in))
3.  The nodes for cancel and continue item exists on the screen, but they are flattened and positioned to the left and below, but still not clickable using either the studio or the yaml file.

### Actual results

The nodes for cancel and continue item exists on the screen, but they are flattened and positioned to the left and below, but still not clickable using either the studio or the yaml file

### Expected results

The nodes for cancel and continue item exists on the screen  and they are  clickable using either the studio or the yaml file.

### About app

- Tested on simulator: Ipad Pro 11-inch (M4)
- This popup is called by native code but invoked from Flutter using platform channel [ex. google_sign_in package](https://pub.dev/packages/google_sign_in)

### About environment

- i am on M3 Pro mac, Sonoma 14.6, 
- Flutter 3.24.3 

### Logs

<details>
<summary>Logs</summary>

```
<!-- Replace this line with your logs. *DO NOT* remove the backticks! -->
```

</details>

### Maestro version

1.39-dev.2

### How did you install Maestro?

Homebrew

### Anything else?

![image](https://github.com/user-attachments/assets/5e9dd994-531b-42f2-b9c7-4ef6bfe01402)
![image](https://github.com/user-attachments/assets/ffd19b0d-3b91-4dc5-bd9c-48dabd84ad03)

**Comment by sanderdegelder** [comment]
I experience the same issue. Our app is landscape only. That works fine with Maestro Studio, but when a native popup of iOS is shown, Maestro Studio misplaces all the elements.

**Comment by Fishbowler** [comment]
Is anyone able to provide a reproducible example?

**Comment by sanderdegelder** [comment]
To reproduce:

- Start app under test in landscape orientation. Our native iOS app only supports landscape orientation.
- Start Maestro Studio.
- Observe that all elements are shown correct and are clickable in Maestro Studio.
- Continue until a native iOS popup is shown. For instance the popup to ask if the app is allowed to track activity across other companies' apps and websites.
- Observe that the screen orientation in Maestro Studio changes to portrait mode and the clickable elements do not match with what is shown on screen.
- <img width="128" alt="Image" src="https://github.com/user-attachments/assets/e906fb43-3b68-42a2-8805-75565fed04ec" />

Expected that the screen orientation stays in landscape mode in Maestro Studio when a native iOS popup is shown.

**Comment by sanderdegelder** [comment]
A more usable example is with the Maps app.

- Make sure location services are disabled.
- Set the screen orientation of the simulator to landscape.
- Start the Maps app.
- Wait for the native iOS popup to to ask permission to use your location.
- Observe that the screen orientation in Maestro Studio changes to portrait mode and the clickable elements do not match with what is shown on screen.

**Comment by sanderdegelder** [comment]
It's getting worse since the release of Maestro 1.41.0.
Now all elements of our app are flattened and positioned to the left and below and they are not recognised by Maestro anymore.
When reverting back to 1.40.3, it works as expected again.

**Comment by Moelfarri** [comment]
Can confirm we have the same issue with Maestro 1.41.0 not properly positioning the semantics so even though they are detected - actual taps don't end up hitting buttons/textfields because of the positioning issue.

**Comment by Fishbowler** [comment]
How's this behaving in the new Maestro Studio desktop app?

**Comment by sanderdegelder** [comment]
Same behaviour in Maestro Studio Desktop as with 1.41.0. Elements are not recognised and if recognised, not tapped on.

**Comment by johntmcintosh** [comment]
@Fishbowler I saw the 2.0.0-dev announcement today so I went to try our landscape flows with the new `setOrientation` command. I too found that basic landscape support had a regression in 1.41.0, and also verified that it was working in 1.40.3. Both through the CLI and the Studio Desktop app. 

I checked out the repo and confirmed from running locally that the regression was introduced in https://github.com/mobile-dev-inc/Maestro/pull/2473

**Comment by mahmoudalaa97** [comment]
Do we have any updates regarding this issue? I faced the same problem when I tried testing on an iPad in landscape mode using Maestro Studio.

**Comment by leggomuhgreggo** [comment]
Also running into this.

**Use case:** Kiosk App; configured with `orientation: 'landscape'`

**Additional Context:** Exploring Maestro adoption; Trying to prove our "local dev" DX with Maestro Desktop

———

<details>
  <summary>Additional Issue Details</summary>

Error
```
Property setOrientation is not allowed.yaml-schema: action(513)
```

<img width="610" height="159" alt="Image" src="https://github.com/user-attachments/assets/f70e85e1-bdaf-49a6-b07d-bb7e18474fb4" />


<img width="1191" height="1462" alt="Image" src="https://github.com/user-attachments/assets/14a8c601-6fd8-459b-8e75-5dfed2fed0e0" />

</details>

**Comment by sanderdegelder** [comment]
Are there any updates on this issue? We are unable to run Maestro tests on our iOS app past version 1.40.3 because of this issue.

**Comment by BobMini** [comment]
Any update here?

**Comment by Fishbowler** [comment]
There was a prospective PR, but it had other side effects that meant it couldn't progress further

**Comment by yolpsoftware** [comment]
Related to #2701?

**Comment by sanderdegelder** [comment]
Yes, what is described [here](https://github.com/mobile-dev-inc/Maestro/issues/2701#issuecomment-3850070869) is exactly the same with what we are experiencing.
